Made minor changes on principles section part 1 (#122)
axelerant/engg-handbook: content/docs/principles/build-for-people.md

---
weight: 2
description: We don't write code in a vacuum. We write code to solve problems for our customers using systems that their customers use, all of which will be maintained by our teammates.
---

# Build for People

We don't write code in a vacuum. We write code to solve problems for our customers using systems that their customers use, all of which will be maintained by our teammates. We write code once but it is read many times and executed many times more than it is read.

## Solve the problem {#solve}

> We write the most beautiful, most efficient, most poetic code you possibly can, but if it doesn't help our customer, it doesn't help anyone. We write great code and solve the problem.

Reviewer: ```suggestion
You could write the most beautiful, most efficient, most poetic code you possibly can, but if it doesn't help our customer, it doesn't help anyone. We write great code and solve the problem.
```

I am reverting part of the change here because the first part of the sentence is intentionally speculative and grandiose. It's not meant to be a fact.

## Write readable code {#readable-code}

> We do not try to be clever with our code. There is a saying that debugging is harder than writing code. So if we write the cleverest possible code then, by definition, we are not clever enough to debug it.

Reviewer: ```suggestion
Do not try to be clever while writing code. There is a saying that debugging is harder than writing code. So if we write the cleverest possible code then, by definition, we are not clever enough to debug it.
```

Similar reason as above.
